### ISSUE TITLE
How to train YOLO in a different way instead of using '.yaml' file ?

### ISSUE THREAD

**Issue by wzh506** [ultralytics/ultralytics#14834]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I don't want to train by using train and .yaml files. I currently want to do a dynamic occlusion task and hope to use yolo for self-supervised training. So the code is as follows:
`            self.map_generator= YOLO("yolov8n-seg.pt")
            print('Make visibility map v3 : yolov8')
            for name, param in self.map_generator.named_parameters():
                param.requires_grad = True`
and I already add parameters to optimizer l:
`self.optimizer = torch.optim.Adam(l, lr=0.0, eps=1e-15)`
But by printing the model weights after the optimizer,
`                optimizer.step()
                optimizer.zero_grad(set_to_none = True)
                after_update = {name: param.clone() for name, param in map_generator.named_parameters()}
                if iteration > opt.masks_iters:
                    for name in before_update:
                        print(f"{name}: ")
                        print(f"Before update: {before_update[name]}")
                        print(f"After update: {after_update[name]}")
                        print(f"Difference: {after_update[name] - before_update[name]}")  `
I found Difference is always zero,I wonder why?

**Comment by pderrenger** [comment]
@wzh506 to train YOLO without using a `.yaml` file, you can directly manipulate the model and optimizer in your code. However, it seems like the issue you're encountering is that the model weights are not updating during training. This could be due to several reasons, such as the learning rate being set to zero or the gradients not being properly calculated.

First, ensure that your learning rate is set to a non-zero value. Additionally, verify that the loss is being computed and backpropagated correctly before calling `optimizer.step()`. If the gradients are zero, the weights will not update.

If the issue persists, please try updating to the latest version of the Ultralytics package to ensure compatibility and bug fixes. If the problem continues, providing a minimal reproducible example would help in diagnosing the issue further.
